### PR TITLE
Updates to maintain compat with Tornado >= 4.5

### LINKED ISSUE #6152: Bokeh Server is incompatible with tornado=4.5
There's a currently undiagnosed issue where Bokeh Server applications are incompatible with the newest tornado v4.5 release.

The server appears to start normally, but doesn't receive any requests or open any sessions when queried.

To reproduce:
* Install Bokeh v0.12.5 via conda or pip
* Run a Bokeh Server app example via ``bokeh serve app.py``
(below is an example app)
* Try (and fail) to open an application session at http://localhost:5006/line_on_off in a browser or via cURL. You'll see that any request just hangs.

contents of app.py
```python
""" Example demonstrating turning lines on and off - with bokeh server

"""

import numpy as np

from bokeh.io import curdoc
from bokeh.layouts import row
from bokeh.palettes import Viridis3
from bokeh.plotting import figure
from bokeh.models import CheckboxGroup

p = figure()
props = dict(line_width=4, line_alpha=0.7)
x = np.linspace(0, 4 * np.pi, 100)
l0 = p.line(x, np.sin(x), color=Viridis3[0], legend="Line 0", **props)
l1 = p.line(x, 4 * np.cos(x), color=Viridis3[1], legend="Line 1", **props)
l2 = p.line(x, np.tan(x), color=Viridis3[2], legend="Line 2", **props)

checkbox = CheckboxGroup(labels=["Line 0", "Line 1", "Line 2"], active=[0, 1, 2], width=100)

def update(attr, old, new):
    l0.visible = 0 in checkbox.active
    l1.visible = 1 in checkbox.active
    l2.visible = 2 in checkbox.active

checkbox.on_change('active', update)

layout = row(checkbox, p)
curdoc().add_root(layout)
```

Platform: MacOS and Ubuntu 14.04
Bokeh = 0.12.5
Tornado = 4.5

### PR DESCRIPTION
**Pull request by bryevdv**
issues: fixes #6152

Explicitly check for None and do not rely on BadYieldError being raised

### CODE REVIEW

File: bokeh/util/tornado.py [hunk 1]
```diff
@@ -16,13 +16,20 @@ def yield_for_all_futures(result):
...
     while True:
+
+        # This is needed for Tornado >= 4.5 where convert_yielded will no
+        # longer raise BadYieldError on None
+        if result is None:
+            break
+
         try:
             future = gen.convert_yielded(result)
         except gen.BadYieldError:
```
Comment (canavandl) [inline bokeh/util/tornado.py]: will this exception ever get raised now?

Comment (bdarnell) [inline bokeh/util/tornado.py]: Yes, for example if `result` is an int or a string.

**Comment by bryevdv** [comment]
Tests are green. @canavandl would suggest merging this and making a dev build tomorrow (I will be out most of the day)

**Comment by bryevdv** [comment]
(merging after review/testing... seems good from what I have tried but more is always better)

**Comment by havocp** [comment]
lgtm!

**Comment by canavandl** [comment]
merged. thanks @bdarnell for the help

**Pull request merged**